# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: f466599094b7
# Full Hash: f466599094b7e16948794a49b0571e03eb2b4df9
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-06-15 13:49:48
# Regressor Bug: 1962133
# File Overlap Count: 14
# Description:
#   Revert "Bug 1962133 - QM: Add accessed flag to FullOriginMetadata and persist it in the origin metadata file; r=dom-storage-reviewers,hsingh" as requested by dmeehan for causing top crash in Bug 1972221.
#   
#   This reverts commit 433768e64dd71c996694e08005854756b7c45fb3.
# ==============================================================================

diff -r 2408bee8bcc4 -r f466599094b7 dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Sat Jun 14 05:48:43 2025 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Sun Jun 15 16:46:59 2025 +0300
@@ -127,7 +127,6 @@
 #include "mozilla/dom/quota/ConditionalCompilation.h"
 #include "mozilla/dom/quota/DirectoryLock.h"
 #include "mozilla/dom/quota/DirectoryLockInlines.h"
-#include "mozilla/dom/quota/DirectoryMetadata.h"
 #include "mozilla/dom/quota/DecryptingInputStream_impl.h"
 #include "mozilla/dom/quota/EncryptingOutputStream_impl.h"
 #include "mozilla/dom/quota/ErrorHandling.h"
@@ -13390,7 +13389,7 @@
     // Loop over "<origin>/idb" directories.
     QM_TRY(CollectEachFile(
         *persistenceDir,
-        [this, &quotaManager, persistenceType, persistent, &idbDirName](
+        [this, &quotaManager, persistenceType, &idbDirName](
             const nsCOMPtr<nsIFile>& originDir) -> Result<Ok, nsresult> {
           if (NS_WARN_IF(QuotaClient::IsShuttingDownOnNonBackgroundThread()) ||
               IsAborted()) {
@@ -13481,20 +13480,6 @@
                   }));
 
               if (!databasePaths.IsEmpty()) {
-                if (!persistent) {
-                  auto maybeOriginStateMetadata =
-                      quotaManager->GetOriginStateMetadata(metadata);
-
-                  auto originStateMetadata = maybeOriginStateMetadata.extract();
-
-                  originStateMetadata.mAccessed = true;
-
-                  QM_TRY(MOZ_TO_RESULT(SaveDirectoryMetadataHeader(
-                      *originDir, originStateMetadata)));
-
-                  quotaManager->UpdateOriginAccessed(metadata);
-                }
-
                 mDirectoryInfos.EmplaceBack(persistenceType, metadata,
                                             std::move(databasePaths));
               }